### PR TITLE
system-refactor

### PR DESCRIPTION
**Pull request by rolfvdhulst**

### CODE REVIEW

**Comment by JaroKuiken** [comment]
@rolfvdhulst Do we want to merge this into development and not master? Wasnt the master branch of _suite the branch that ended up working for Max and others in the end? And i see that there are 4 commits by Johan on Jul 21 that are on master but not on development, while development has 1 commit by Johan on Sep 9 that master doesnt have. Im not sure if those 4 or that 1 commit arent/isnt necessary? If one of them isnt necessary, then we can use the other branch. But if they are necessary, we should pick a branch and also put those commits into that branch.

**Comment by JaroKuiken** [comment]
Once we figure that out, this can also be merged imo

**Review by Yuhanun — APPROVED** [review]
LGTM looks like CMake magic to me

**Comment by rolfvdhulst** [comment]
@JaroKuiken Good point, I did not notice. I think we should probably merge master into development then, and then make development the default branch.

**Comment by rolfvdhulst** [comment]
I updated development with master

**Review by JaroKuiken — APPROVED** [review]

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+MaxEmptyLinesToKeep: '2'
+NamespaceIndentation: All
+SpaceBeforeAssignmentOperators: 'true'
+SpaceBeforeParens: ControlStatements
+SpaceInEmptyParentheses: 'false'
+SpacesInAngles: 'false'
+SpacesInCStyleCastParentheses: 'false'
+SpacesInContainerLiterals: 'true'
+SpacesInParentheses: 'false'
+Standard: Cpp11
```
Comment (Yuhanun) [inline .clang-format]: should probably be 20

File: checkout_branch.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    
+    raise Exception("Failed to checkout repo")
+
+def checkout_main_repos_branch(branch: str):
+    main_repos: List[str] = [
+        "roboteam_ai",
+        "roboteam_proto",
+        "roboteam_robothub",
+        "roboteam_utils",
+        "roboteam_world"
```
Comment (Yuhanun) [inline checkout_branch.py:19]:
```suggestion
        "roboteam_world",
        "roboteam_central_server"
```

**Pull request merged**